### PR TITLE
[protractor-beautiful-reporter] Add types

### PR DESCRIPTION
**Pull request by akwiatek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## protractor-beautiful-reporter ([<kbd>unpkg</kbd>](https://unpkg.com/browse/protractor-beautiful-reporter@latest/))
was missing the following properties:
1. spaces
1. readFile
1. readFileSync
1. writeFile
1. writeFileSync

<details>
<summary>as well as these 17 other properties...</summary>
<p>f, fromCallback, fromPromise, BREAK, RETURN, exists, getRootPath, invalidWin32Path, symlinkPaths, symlinkPathsSync, symlinkType, symlinkTypeSync, moveSync, hasMillisRes, hasMillisResSync, timeRemoveMillis, utimesMillis</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7bfd7bdd94d4ee786eda0cb0de430a7954559e23
</p>

File: types/protractor-beautiful-reporter/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+export = HtmlReporter;
+
+declare class HtmlReporter {
+    constructor(options: HtmlReporter.HtmlReporterConstructorOptions);
+    getJasmine2Reporter(): jasmine.CustomReporter;
+}
+
+declare namespace HtmlReporter {
+    interface HtmlReporterConstructorOptions {
```
Comment (peterblazejewicz) [inline types/protractor-beautiful-reporter/index.d.ts:16]: should it accept `jasmine2MetaDataBuilder` option?:
https://github.com/Evilweed/protractor-beautiful-reporter#jasmine2-meta-data-builder-optional

File: types/protractor-beautiful-reporter/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        baseDirectory: string;
+        pathBuilder?: (
+            spec: null,
+            descriptions: string[],
+            result: jasmine.SpecResult,
+            capabilities: Capabilities,
+        ) => string;
```
Comment (akwiatek) [inline types/protractor-beautiful-reporter/index.d.ts:30]: Based on https://github.com/Evilweed/protractor-beautiful-reporter/blob/c2c22a71eabff89ea6d8b08c1b77dbf84edd86cd/index.js#L4762

Please note that it says that spec is an object but actually null specs are passed to both builder functions
https://github.com/Evilweed/protractor-beautiful-reporter/blob/c2c22a71eabff89ea6d8b08c1b77dbf84edd86cd/app/reporter.js#L271

File: types/protractor-beautiful-reporter/index.d.ts (hunk 1 above)
Comment (akwiatek) [reply types/protractor-beautiful-reporter/index.d.ts:23]: Done

File: types/protractor-beautiful-reporter/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import jasmine = require('jasmine');
+import { Capabilities } from 'selenium-webdriver';
+
+export = HtmlReporter;
+
+declare class HtmlReporter {
+    constructor(options: HtmlReporter.HtmlReporterConstructorOptions);
+    getJasmine2Reporter(): jasmine.CustomReporter;
+    jasmine2MetaDataBuilder(
+        spec: null,
```
Comment (akwiatek) [inline types/protractor-beautiful-reporter/index.d.ts:15]: According to https://github.com/Evilweed/protractor-beautiful-reporter/blob/c2c22a71eabff89ea6d8b08c1b77dbf84edd86cd/app/reporter.js#L271

pathBuilder(), jasmine2MetaDataBuilder() always get null specs.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@akwiatek thanks!

**Comment by akwiatek** [comment]
Ready to merge

**Pull request merged**